wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wdl/WdlWriterImpl.scala

```diff
@@ -0,0 +1,438 @@
+          pair._1 + ": " + expressionElementWriter.toWdlV1(pair._2)
+        }.mkString(", ") + " }"
+      case a: ArrayLiteral =>
+        "[" + a.elements.map(expressionElementWriter.toWdlV1).mkString(", ") + "]"
+      case a: MapLiteral =>
+        "{ " + a.elements.map { pair =>
+          expressionElementWriter.toWdlV1(pair._1) + ": " + expressionElementWriter.toWdlV1(pair._2)
+        }.mkString(", ") + " }"
+      case a: PairLiteral =>
+        s"(${expressionElementWriter.toWdlV1(a.left)}, ${expressionElementWriter.toWdlV1(a.right)})"
+      case a: UnaryOperation => a.toWdlV1
+      case a: BinaryOperation => a.toWdlV1
+      case a: TernaryIf =>
+        s"if ${expressionElementWriter.toWdlV1(a.condition)} then ${expressionElementWriter.toWdlV1(a.ifTrue)} else ${expressionElementWriter.toWdlV1(a.ifFalse)}"
+      case a: FunctionCallElement => a.toWdlV1
+      case a: IdentifierLookup => a.identifier
+      case a: IdentifierMemberAccess => a.toWdlV1
+      case a: ExpressionMemberAccess => s"${expressionElementWriter.toWdlV1(a.expression)}.${a.memberAccessTail.toList.mkString(".")}"
+      case a: IndexAccess => s"${expressionElementWriter.toWdlV1(a.expressionElement)}[${expressionElementWriter.toWdlV1(a.index)}]"
+    }
+  }
+
+  implicit val unaryOperationWriter: WdlWriter[UnaryOperation] = new WdlWriter[UnaryOperation] {
+    override def toWdlV1(a: UnaryOperation): String = a match {
+      case a: LogicalNot    => s"!(${a.argument.toWdlV1})"
+      case a: UnaryNegation => s"-(${a.argument.toWdlV1})"
+      case a: UnaryPlus     => s"+(${a.argument.toWdlV1})"
+    }
+  }
+
+  implicit val identifierMemberAccessWriter: WdlWriter[IdentifierMemberAccess] = new WdlWriter[IdentifierMemberAccess] {
+    override def toWdlV1(a: IdentifierMemberAccess): String = {
+      s"${a.first}.${a.second}" + (if (a.memberAccessTail.nonEmpty) {
+        "." + a.memberAccessTail.mkString(".")
+      } else {
+        ""
+      })
+    }
+  }
+
+  implicit val binaryOperationWriter: WdlWriter[BinaryOperation] = new WdlWriter[BinaryOperation] {
+    override def toWdlV1(a: BinaryOperation) = {
+      def sandwich(opString: String) = s"(${a.left.toWdlV1} $opString ${a.right.toWdlV1})"
+
+      a match {
+        case _: LogicalOr           => sandwich("||")
+        case _: LogicalAnd          => sandwich("&&")
+        case _: Equals              => sandwich("==")
+        case _: NotEquals           => sandwich("!=")
+        case _: LessThan            => sandwich("<")
+        case _: LessThanOrEquals    => sandwich("<=")
+        case _: GreaterThan         => sandwich(">")
+        case _: GreaterThanOrEquals => sandwich(">=")
+        case _: Add                 => sandwich("+")
+        case _: Subtract            => sandwich("-")
+        case _: Multiply            => sandwich("*")
+        case _: Divide              => sandwich("/")
+        case _: Remainder           => sandwich("%")
+      }
```

Comment:
You should be able to DRY this out a bunch by only returning the String from the `match` and then `sandwich`ing that. But then less sandwiches.